### PR TITLE
Added some benchmarks guide to DEVGUIDE.md

### PR DESCRIPTION
**Pull request by vzarytovskii**
Added simple FCS/FSharp.Core benchmarking guide to the [DEVGUIDE.md](DEVGUIDE.md).
I would appreciate if someone tries to follow steps described and see if there are any issues.

**Update:** rendered version can be found here: https://github.com/dotnet/fsharp/blob/04578cb898b1704cc0d0ec5d890c19f369337f9c/DEVGUIDE.md

Thanks.

### CODE REVIEW

File: DEVGUIDE.md [hunk 1]
```diff
@@ -173,6 +180,160 @@ It's important to use `Release` if you want to see if your changes have had a no
 
 Use the `Debug` configuration to test your changes locally. It is the default. Do not use the `Release` configuration! Local development and testing of Visual Studio tooling is not designed for the `Release` configuration.
 
+#### Writing and running benchmarks
```
Comment (rstm-sf) [inline DEVGUIDE.md:183]: Maybe it's better to change the levels? Headlines are hard to read

image

Comment (vzarytovskii) [reply DEVGUIDE.md]: I've changed structure a bit, bumped some headings, should be easier to read now.

File: DEVGUIDE.md [hunk 2]
```diff
@@ -188,6 +191,164 @@ To fix this, delete these folders:
...
+### Benchmarking and profiling the compiler
+
+**NOTE:** When running benchmarks or profiling compiler, and comparing results with upstream version, make sure:
+
+* Always build both versions of compiler/FCS from source and not use pre-built binaries from SDK (SDK binaries are crossgen'd, which can affect performance).
+* To run `Release` build of compiler/FCS.
+
+### Example benchmark setup using [BenchmarkDotNet](https://github.com/dotnet/BenchmarkDotNet)
+
+1. Perform a clean build of the compiler and FCS from source(as described in this document).
```
Comment (nojaf) [inline DEVGUIDE.md:211]: Would that be `.\Build.cmd -noVisualStudio`, or do you need some other bits as well?

File: DEVGUIDE.md [hunk 3]
```diff
@@ -188,6 +191,164 @@ To fix this, delete these folders:
...
+Existing compiler benchmarks can be found in `tests\benchmarks\`.
+
+### Benchmarking and profiling the compiler
+
+**NOTE:** When running benchmarks or profiling compiler, and comparing results with upstream version, make sure:
+
+* Always build both versions of compiler/FCS from source and not use pre-built binaries from SDK (SDK binaries are crossgen'd, which can affect performance).
+* To run `Release` build of compiler/FCS.
+
+### Example benchmark setup using [BenchmarkDotNet](https://github.com/dotnet/BenchmarkDotNet)
```
Comment (nojaf) [inline DEVGUIDE.md:209]: Maybe also add a note about what expectations users can have here.
Is this more meant as how would you do it for your local project?
Or more meant as how do I contribute samples?

Comment (vzarytovskii) [reply DEVGUIDE.md:209]: Actually, both. Benchmarks and/or results can be committed as part of PR, for example.

File: DEVGUIDE.md (hunk 2 above)
Comment (vzarytovskii) [reply DEVGUIDE.md]: Added a note there.

File: DEVGUIDE.md (hunk 3 above)
Comment (vzarytovskii) [reply DEVGUIDE.md:209]: I've added a note for that.

**Pull request merged**